proxy/envoy/ingress_test.go

```diff
@@ -44,16 +48,23 @@ func TestRouteCombination(t *testing.T) {
 		{path3, prefix1, nil},
 		{path3, prefix2, nil},
 		{path3, prefix3, path3},
+		{path1, regex1, path1},
+		{prefix1, regex1, prefix1},
```

Comment:
Shouldn't this be `path1`? Regex only matches `/xyz` or `/xya`

Reply:
Yes, the prefix1 is /xyz

Reply:
Oh I see what you mean. But we can't know ahead of time. prefix matching a regex does not mean we convert the prefix into an exact path, because that prefix might be an invalid path

Reply:
The test is incorrect then. Please fix the test and the logic. I suggest converting prefix into a regex.